### ISSUE TITLE
Can't change the name of a blueprint

### ISSUE THREAD

**Issue by jdanish** [theRAPTLab/gsgo#767]
To reproduce: click on a script / blueprint name. Select the top line. Change the text and press save line. Screen goes white and the following errors appear. This is happening in the 2 most recent branches (f64725295d0d063b48a5e6ab188fe08bb7ce275b and f12aee8af7a57ce291f3d08fd6bc3d878daad8ec)

<img width="1493" alt="Screenshot 2023-08-30 at 7 49 42 PM" src="https://github.com/theRAPTLab/gsgo/assets/5529898/4ae00319-5504-4873-851f-4e8e457f57cc">

**Comment by benloh** [comment]
@jdanish The basic problem here is that changing the name of the `# BLUEPRINT ...` pragma still leaves a ton of references to the original blueprint name, so the system is choking on the bad blueprint name.

e.g. if you change `# BLUEPRINT Beaver` to `# BLUEPRINT Badger`, you still have a ton of lines like...
```
featProp Beaver.Costume.costumeName setTo 'WL_beaver.png'
featProp Beaver.Costume.scale setTo 0.15
prop Beaver.zIndex setTo 200
```
..that still point to `Beaver`.

I suppose the proper solution is to do a full code search and replace.  I could've sworn we've implemented this before but it's possible we only did it for the code view and not the Wizard view.

**Comment by benloh** [comment]
Thinking this through some more, a few possible approaches:

1. Add a special "Blueprint" edit mode
Instead of treating the `# BLUEPRINT` line as just another Wizard Keyword, we pull it out of the wizard interface and allow to change the name in a special location (e.g. a title bar), automatically replacing the existing name with a new name after you've confirmed the change.  This could get complicated as we don't necessarily want to do a script text search and replace where we can inadvertently change other references to the old blueprint name (e.g. you duplicate a Beaver blueprint, adding a new Badger character and still want to reference Beaver in the code).  The "smart" fix would be to find the code object references to "Beaver" rather than simple string replace.  It's a bit more work but should be possible.
This is probably not worth the trouble.

2. Search and Replace after wizard / code update
If we want to support editing the wizard or code view using the existing tools, then we need to add a review pass after the line is changed to do the search and replace.  So we'd be adding a special code check to see if the user has changed the name, and then attempting to update all the references.   The advantage here is we don't need to build a new UI, but we'd still have to do the "smart" object replace.

3. Use "character"
The js error is stemming from use of the old blueprint name. If you were to change the code references from "Beaver" to "character" (or "agent" if we are unable to fix #767) then I believe the error will go away.  This is a cheap solution that requires no code, just a rewrite of the script.  (Though we might want to add a way to catch the error without breaking.)
This is the simplest solution, but it might make your script a little less readable.

4. Block blueprint name changes
The cheap fix is to basically disallow these changes.  We can block wizard and code view changes to the blueprint name.  So if you wanted to do that kind of change, you'd need to edit the project by hand.  I'm guessing this isn't an option.

5. Do nothing
SInce changing the name results in an error, we probably can't afford just ignore the problem.

I'm guessing (2) is the most likely solution.

**Comment by benloh** [comment]
Addressed with #770
